### PR TITLE
docs(configuration): add explanation of new path definition which includes first/last callback execution

### PR DESCRIPTION
**Pull request by ymski**
## Description

Add the document of Path analysis option, `include_first_callback` and `include_last_callback`.

## Related links
- https://github.com/tier4/CARET_analyze/pull/263

## Pre-review checklist for the PR author

- [x] I've confirmed the [contribution guidelines](https://github.com/tier4/caret/blob/main/.github/CONTRIBUTING.md).
- [x] The PR is ready for review.

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR has been reviewed.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [x] There are no open discussions or they are tracked via tickets.
- [x] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

### CODE REVIEW

File: docs/configuration/inter_node_data_path.md [hunk 1]
```diff
@@ -157,3 +157,32 @@ arch.export('new_architecture.yaml')
...
+In the default configuration, the path analysis does not include the callback processing times at both ends of the above paths. The following two processing times are not included by default in path analysis.
+
+- `callback_start` to `rclcpp_publish` or `rclcpp_intra_publish` in the first node.
+- `callback_start` to `callback_end` in the last node.
+
+We provide options for evaluating the callback times at the end of these paths.
+
+- `Path.include_first_callback`
+  - Default: `False`
+  - If `True`, the time from callback_start to publish is included in the path analysis. The trace data used for the above callback_start adopts the data closest to the time of publish among those with the same tid as publish. Note that the first callback time added here may contain more than one callback object.
```
Comment (takeshi-iwanari) [inline docs/configuration/inter_node_data_path.md:173]:
```suggestion
  - If `True`, the processing time in the first node is included in the path analysis. The callback_start in the first node is that of a trace data whose timestamp is the closest to publish and thread id (tid) is the same as publish. Note that a callback object of the callback_start in the first node can vary if the node has several callbacks and any of them runs before publish.
```

File: docs/configuration/inter_node_data_path.md [hunk 2]
```diff
@@ -157,3 +157,32 @@ arch.export('new_architecture.yaml')
...
+- `Path.include_last_callback`
+  - Default: `False`
+  - If `True`, the time from callback_start to callback_end is included in the path analysis.
```
Comment (takeshi-iwanari) [inline docs/configuration/inter_node_data_path.md:176]:
```suggestion
  - If `True`, the processing time in the last node is included in the path analysis.
```

File: docs/configuration/inter_node_data_path.md [hunk 3]
```diff
@@ -157,3 +157,32 @@ arch.export('new_architecture.yaml')
 ```
 
 An exception is raised for pairs that cannot be combined.
+
+### Considering first/last callback
+
+In CARET, the path is defined as `[node_name]-[topic_name]-... -[topic_name]-[node_name]` (for more information in [Path](../design/event_and_latency_definitions/path.md#Path)).
+In the default configuration, the path analysis does not include the callback processing times at both ends of the above paths. The following two processing times are not included by default in path analysis.
+
+- `callback_start` to `rclcpp_publish` or `rclcpp_intra_publish` in the first node.
```
Comment (takeshi-iwanari) [inline docs/configuration/inter_node_data_path.md:166]:
```suggestion
- Processing time in the first node; from `callback_start` to `rclcpp_publish` or `rclcpp_intra_publish`
```

File: docs/configuration/inter_node_data_path.md [hunk 4]
```diff
@@ -157,3 +157,32 @@ arch.export('new_architecture.yaml')
...
+- `callback_start` to `callback_end` in the last node.
```
Comment (takeshi-iwanari) [inline docs/configuration/inter_node_data_path.md:167]:
```suggestion
- Processing time in the last node; from `callback_start` to `callback_end`
```

**Review by takeshi-iwanari — CHANGES_REQUESTED** [review]
I added some suggestions. Please adopt them if you like it.

**Comment by ymski** [comment]
@takeshi-iwanari 
Thank you for your review. The explanations you proposed were good, so I have adopted them as they are.

**Review by takeshi-iwanari — APPROVED** [review]
LGTM

**Comment by takeshi-iwanari** [comment]
@ymski 
Thank you for working on it.
You can merge this PR by yourself.

**Pull request merged**